src/mds/MDLog.h

```diff
@@ -301,5 +306,16 @@ class MDLog {
   std::set<LogSegment*> expired_segments;
   std::set<LogSegment*> expiring_segments;
   uint64_t events_since_last_major_segment = 0;
+
+  // log trimming decay counter
+  DecayCounter log_trim_counter;
+
+  // log trimming upkeeper thread
+  std::thread upkeep_thread;
+  // lock order
+  // -> mds_lock
+  //        -> submit_mutex
```

Comment:
this comment isn't relevant here anymore. Also, the submit mutex may be locked directly, just never before the mds_lock if both need to be taken.

Reply:
So, this comments is best moved elsewhere, but its still holds valid since the locking rule is detailed. I thought about that and kept it as-it-is.

Reply:
OK, let's move it to the `submit_mutex` declaraition then.